test/functional/zmq_test.py

```diff
@@ -13,9 +13,33 @@
                                  hash256,
                                 )
 
+class ZMQSubscriber:
+    def __init__(self, socket, address):
+        self.address = address
+        self.socket = socket
+        self.sequence = 0
+
 class ZMQTest (BitcoinTestFramework):
     def set_test_params(self):
+        self.address = "tcp://127.0.0.1:28332"
+        self.num_blocks = 5
         self.num_nodes = 2
+        self.subscribers = {}
+
+    def subscribe(self, type):
```

Comment:
As in my comment here https://github.com/bitcoin/bitcoin/pull/11439#discussion_r142474301, I think it makes sense for the subscribe and receive logic to be owned by `ZMQSubscriber`

Reply:
Ok.